iconservice/icx/icx_engine.py

```diff
@@ -218,14 +218,16 @@ def get_balance(self,
         return account.balance
 
     # todo: need to be refactoring
-    def get_total_supply(self, context: 'IconScoreContext') -> int:
+    def get_total_supply(self) -> int:
         """Get the total supply of icx coin
 
-        :param context:
         :return: (int) amount in loop (1 icx == 1e18 loop)
         """
-        # todo: to be refactored (using memory)
-        return self._storage.get_total_supply(context)
+
+        return self._total_supply_amount
+
+    def set_total_supply(self, total_supply: int):
```

Comment:
please, add assert total_supply > new_total_supply

Reply:
I think this would be good to transform to "property" or rename nice.
because it uses only for issue.

but I have no idea now.